modules/3d/src/ptcloud/sampling.cpp

```diff
@@ -278,30 +245,88 @@ int farthestPointSampling(OutputArray sampled_point_flags, InputArray input_pts,
+
+#ifdef CV_SIMD
+    // Ensure that the point(in C) data x,y,z is continuous in the memory respectively
+    _swap(ori_pts_ptr_x[seed], ori_pts_ptr_x[0]);
+    _swap(ori_pts_ptr_y[seed], ori_pts_ptr_y[0]);
+    _swap(ori_pts_ptr_z[seed], ori_pts_ptr_z[0]);
+#endif
 
-    float *const ori_pts_ptr = (float *) ori_pts.data;
     int sampled_cnt = 1;
     const float dist_lower_limit_square = dist_lower_limit * dist_lower_limit;
     while (sampled_cnt < sampled_pts_size)
     {
+        // The position of the point that was last added to S
+#ifdef CV_SIMD
         int last_pt = sampled_cnt - 1;
-        float *const last_pt_ptr_base = ori_pts_ptr + 3 * idxs[last_pt];
-        float last_pt_x = last_pt_ptr_base[0], last_pt_y = last_pt_ptr_base[1], last_pt_z = last_pt_ptr_base[2];
+#elif
+        int last_pt = idxs[sampled_cnt - 1];
+#endif
+        float last_pt_x = ori_pts_ptr_x[last_pt];
+        float last_pt_y = ori_pts_ptr_y[last_pt];
+        float last_pt_z = ori_pts_ptr_z[last_pt];
 
         // Calculate the distance from point in C to set S
         float max_dist_square = 0;
+
+#ifdef CV_SIMD
+        // Calculate how many floats can be stored in each vector
+        int step = sizeof(v_float32) / sizeof(float);
+        int k = (ori_pts_size - sampled_cnt) / step;
+        int end = sampled_cnt + step * k;
+
+        std::vector<float> last_p_x_arr(step, last_pt_x);
+        v_float32 v_last_p_x = vx_load(last_p_x_arr.data());
+        std::vector<float> last_p_y_arr(step, last_pt_y);
+        v_float32 v_last_p_y = vx_load(last_p_y_arr.data());
+        std::vector<float> last_p_z_arr(step, last_pt_z);
+        v_float32 v_last_p_z = vx_load(last_p_z_arr.data());
+
+        AutoBuffer<float> _next_dist_square_arr(step);
+        float *next_dist_square_arr = _next_dist_square_arr.data();
+        for (int i = sampled_cnt; i < end; i += step)
+        {
+            v_float32 vx_diff = v_last_p_x - vx_load(ori_pts_ptr_x + i);
+            v_float32 vy_diff = v_last_p_y - vx_load(ori_pts_ptr_y + i);
+            v_float32 vz_diff = v_last_p_z - vx_load(ori_pts_ptr_z + i);
+
+            v_float32 v_next_dist_square =
+                    vx_diff * vx_diff + vy_diff * vy_diff + vz_diff * vz_diff;
+
+            vx_store(next_dist_square_arr, v_next_dist_square);
+            for (int m = 0; m < step; ++m)
+            {
+                int ds_idx = i + m;
+                if (next_dist_square_arr[m] < dist_square[ds_idx])
+                {
+                    dist_square[ds_idx] = next_dist_square_arr[m];
+                }
+                if (dist_square[ds_idx] > max_dist_square)
+                {
+                    last_pt = ds_idx;
+                    max_dist_square = dist_square[ds_idx];
+                }
+            }
+
+        }
+
+        for (int i = end; i < ori_pts_size; ++i)
+#else
         for (int i = sampled_cnt; i < ori_pts_size; ++i)
+#endif
         {
```

Comment:
Please check code pattern if existed SIMD loops:

```
    int i = sampled_cnt;
#ifdef CV_SIMD
    ...
    for (; i <= end; i += step)
    {
        ...
    }
    // i = end;  // usually this line not needed at all (for loop exits on this condition)
#endif
    for (; i < ori_pts_size; ++i)
    {
        ... generic C++ processing ...
    }
```

Reply:
Got it~